Source/CBLDatabaseUpgrade.m

```diff
@@ -163,7 +163,70 @@ - (CBLStatus) moveAttachmentsDir {
             return CBLStatusFromNSError(error, 0);
         }
     }
-    return kCBLStatusOK;
+
+    return [self renameAttachmentFileNamesInDir: newAttachmentsPath];
+}
+
+
+- (CBLStatus) renameAttachmentFileNamesInDir: (NSString*)dir {
+    // CBL Android 1.0.4 and .NET 1.1.0 have lowercase attachment file names.
+    // This method will detect that and uppercase the file names if needed.
+    NSFileManager* fmgr = [NSFileManager defaultManager];
+
+    if (![fmgr fileExistsAtPath: dir])
+        return kCBLStatusOK;
+
+    NSError* error;
+    NSArray* content = [fmgr contentsOfDirectoryAtPath: dir error: &error];
+    if (error)
+        return CBLStatusFromNSError(error, kCBLStatusAttachmentError);
+
+    BOOL success = YES;
+    NSMutableArray* processed = [NSMutableArray array];
+    for (NSString* fileName in content) {
+        if (![[fileName pathExtension] isEqualToString:@"blob"])
+            continue;
+
+        NSString* filePath = [dir stringByAppendingPathComponent: fileName];
+        NSString* newPath = [self uppercaseAttachmentFileNameAtPath: filePath];
```

Comment:
This code appends the filename to the path, extracts the filename back out of the path, uppercases it, truncates the filename from the path, and adds the new filename. Couldn't you just uppercase the filename _before_ adding it to the path?